src/wallet/wallet.cpp

```diff
@@ -1925,37 +1991,81 @@ bool CWallet::CreateTransaction(const vector<pair<CScript, int64_t> >& vecSend,
                     wtxNew.vout.insert(position, CTxOut(nChange, scriptChange));
                 }
                 else
+                {
                     reservekey.ReturnKey();
+                }
 
-                // Fill vin
-                for (auto const& coin : setCoins)
-                    wtxNew.vin.push_back(CTxIn(coin.first->GetHash(),coin.second));
+                if (setCoins_in.size())
```

Comment:
nit: `auto setCoins = provided_coin_set ? setCoins_in : setCoins_out; `

Reply:
Yes. Looks prettier but results in a copy.